### PR TITLE
snscrape: init at 0.1.3

### PR DESCRIPTION
**Pull request by ivan**
###### Motivation for this change

[snscrape](https://github.com/JustAnotherArchivist/snscrape) is a social networking service scraper that discovers URLs on Twitter, Instagram, Google Plus, and Facebook.

###### Things done

- [x] Tested using sandboxing ([nix.useSandbox](http://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](http://nixos.org/nix/manual/#sec-conf-file) on non-NixOS)
- Built on platform(s)
   - [x] NixOS
   - [x] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nox --run "nox-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [ ] Assured whether relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

---

I manually tested that this works:

```
snscrape twitter-user patio11
```

### CODE REVIEW

**Comment by ivan** [comment]
Updated to test that `snscrape --help` works.

File: pkgs/top-level/all-packages.nix [hunk 1]
```diff
@@ -22816,6 +22816,8 @@ in
 
   snapraid = callPackage ../tools/filesystems/snapraid { };
 
+  snscrape = python3Packages.callPackage ../development/python-modules/snscrape { };
```
Comment (dotlambda) [inline pkgs/top-level/all-packages.nix]:
```suggestion
   snscrape = with python3Packages; toPythonApplication snscrape;
```

File: pkgs/development/python-modules/snscrape/default.nix [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+  # There are no tests; make sure the executable works.
+  checkPhase = ''
+    export PATH=$PATH:$out/bin
+    snscrape --help
+  '';
+
+  propagatedBuildInputs = [ requests lxml beautifulsoup4 ];
+
+  meta = with stdenv.lib; {
```
Comment (dotlambda) [inline pkgs/development/python-modules/snscrape/default.nix]:
```suggestion
   meta = with lib; {
```

File: pkgs/development/python-modules/snscrape/default.nix [hunk 3]
```diff
@@ -0,0 +1,44 @@
+{ stdenv
```
Comment (dotlambda) [inline pkgs/development/python-modules/snscrape/default.nix]:
```suggestion
 { lib
```

File: pkgs/development/python-modules/snscrape/default.nix [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+    inherit pname version;
+    sha256 = "1mnhqqc7xfwg2wrzpj1pjbcisjjwxrgmy21f53p80xbx2iz8b9n1";
+  };
+
+  patches = [ 
+    (fetchpatch {
+      # Avoid (IP, UA)-keyed bans: https://github.com/JustAnotherArchivist/snscrape/pull/26
+      url = "https://github.com/JustAnotherArchivist/snscrape/commit/1984110f785e64d993c391e88f67a8980b743f62.patch";
+      sha256 = "00iagnxdlrlrj3yry6l16xjbra8p0dnl10m6rqhy8rp7i4zdjrmf";
+    })
```
Comment (dotlambda) [inline pkgs/development/python-modules/snscrape/default.nix]: I don't think we should apply this patch before upstream does so.

**Review by dotlambda — CHANGES_REQUESTED** [review]

**Comment by dotlambda** [comment]
Btw, putting this into `pythonPackages` is only allowed because "it is also possible to use snscrape as a library in Python": https://github.com/JustAnotherArchivist/snscrape#usage

**Pull request merged**